models/virtual_service.go

```diff
@@ -57,12 +57,15 @@ func (vService *VirtualService) IsValidHost(namespace string, serviceName string
 	if serviceName == "" {
 		return false
 	}
-	if hosts, ok := vService.Spec.Hosts.([]interface{}); ok {
-		for _, host := range hosts {
-			if kubernetes.FilterByHost(host.(string), serviceName, namespace) {
-				return true
-			}
-		}
+
+	protocolNames := []string{"http", "tcp"}
```

Comment:
I'm not reviewing this one, just looked by change, but I think the protocol names is incomplete as "tls" is also a valid protocol for a VS.

Reply:
We are following what the service details page is doing here, which is only checking for http and tcp (https://github.com/kiali/kiali/blob/master/kubernetes/istio_details_service.go#L79).

If we want to handle tls as well for virtualservices, we probably also need to update how the service details handles things

Reply:
Is there a reason we don't allow https then? Validators do accept it and also validate it.

Reply:
We should rather fix service details than repeat the mistake here also.

Reply:
This PR just brings what the graph is doing in line with that is happening with the detail pages (which just uses http and tcp, see https://github.com/kiali/kiali/blob/master/kubernetes/istio_details_service.go#L188).

If we want to improve that, then lets open another jira to add in support for tls (which is the only other option https://istio.io/docs/reference/config/networking/v1alpha3/virtual-service/#VirtualService).

Reply:
+1 to merge this one and fix the pending tls prototocol in a following PR
@mwringe CI is claiming about some minor style issue, if you can fix it we can move on.